content/post/formatting-with-style.md

```diff
@@ -0,0 +1,257 @@
+- `if age <= 18, sample is classified A`
+
+This way we can only display one rule (or a few rules) and still
+understand (partly) why the model made a decision.
+
+### Evaluation
+
+To evaluate those models, we created articial datasets to make sure
+that we could detect rogue style and propose meaningful fixes. Some of
+those datasets used manually inserted mistakes while others were
+created using simple automatic transformations.
+
+## Meta Learning
+
+As detailed in the previous section on our first approach, we first
+considered methods which only required the target codebase as input to
+work. This already allowed us to train models that addressed the
+painpoint we wanted to annihilate.
+
+Still, as Machine Learning engineers, we are always sad not to use
+data when it is widely available. For this reason we're working on a
+second incarnation of our format analyzer, that works in two distinct
+steps: first it learns to model style on many repositories, then it is
+applied to a given codebase. This is in opposition to our first
+approach, which only did the latter.
+
+To be clear though, our goal is still __not__ to model _global style
+practices_. For this reason, we don't want to use standard supervised
+learning approaches but instead have turned to meta-learning, which
+can be described as the set of techniques that allow to _learn how to
+learn_.
+
+The plan is simple:
+
+1. Learn how to learn to model style on a given repository thanks to
+   all the repositories we can find
+2. Apply this knowledge to model one precise repository
+
+This is quite different from simple supervised learning.
+
+To achieve this meta learning objective, we plan to use known deep
+learning techniques:
+
+### Step 1, learning to learn
+
+The first phase and the most time-consumming one is to train an
+efficient, general model that can learn how to model style in a repo
+given minimal effort.
+
+To do that, we use use the common idea of an _embedding_ to model
+style, with a twist: the embedding is the only place where the model
+can store information about the style of a repository and therefore
+__has__ to condense __all the style information__ into it.
+
+### Step 2, applying the formatter to new repositories
+
+Since the style embedding is the only place where our model can encode
+style information about a repository, we can use techniques that are
+well-known in both Natural Language Processing and Computer Vision
+communities to find the embedding that most closely models the style
+of the repository at hand.
+
+And that's it! Since all of the style information is condensed into
+one single place, we've nothing to do.
+
+### Why does it work?
+
+This simple two-steps process is very efficient because the model used
+can be extremely complex and expressive to use the style embedding to
+produce style hints. It can learn on millions of repositories how
+certain styles are applied in certain corner cases.
```

Comment:
It can learn how certain styles are applied in certain corner cases in millions of repositories.